### ISSUE TITLE
Cannot resolve dependencies when using Image: 3.0.1 

### ISSUE THREAD

**Issue by fotiDim** [aws-amplify/amplify-flutter#427]
**Describe the bug**
In our project we depend both on `Amplify` and the `Image` package. There is currently a transitive dependency conflict.

```
flutter pub get
Running "flutter pub get" in my_awesome_project...     
    Because no versions of image match >3.0.1 <4.0.0 and image 3.0.1 depends on archive ^3.0.0, image ^3.0.1 requires archive ^3.0.0.

(1) So, because archive >=3.0.0 depends on crypto ^3.0.0 and uuid >=2.0.0-rc1 <3.0.0-nullsafety.0 depends on crypto ^2.0.0, image ^3.0.1 is incompatible with uuid >=2.0.0-rc1 <3.0.0-nullsafety.0.



    Because no versions of amplify_flutter match >=0.1.0 <0.1.0-∞ and amplify_flutter <0.0.2-dev.2 depends on amplify_datastore_plugin_interface ^0.0.2-dev.1, amplify_flutter <0.0.2-dev.2 or >=0.1.0-0 <0.1.0 requires amplify_datastore_plugin_interface ^0.0.2-dev.1.

    And because amplify_flutter ^0.0.2-dev.2 depends on amplify_datastore_plugin_interface ^0.0.2-dev.2 and amplify_flutter >=0.1.0 depends on amplify_api_plugin_interface 0.1.0, every version of amplify_flutter requires amplify_datastore_plugin_interface ^0.0.2-dev.1 or amplify_api_plugin_interface 0.1.0.

    And because every version of amplify_datastore_plugin_interface depends on uuid ^2.2.2 and every version of amplify_api_plugin_interface depends on uuid ^2.2.2, every version of amplify_flutter requires uuid ^2.2.2.

    And because image ^3.0.1 is incompatible with uuid >=2.0.0-rc1 <3.0.0-nullsafety.0 (1), image ^3.0.1 is incompatible with amplify_flutter.

    So, because my_awesome_project depends on both image ^3.0.1 and amplify_flutter <1.0.0, version solving failed.
pub get failed (1;     So, because my_awesome_project depends on both image ^3.0.1 and amplify_flutter <1.0.0, version solving failed.)
exit code 1
```
**To Reproduce**
1. Go to 'Pubspec.yaml'
2. Put
```
  image: ^3.0.1
  amplify_flutter: '<1.0.0'  
  amplify_auth_cognito: '<1.0.0'
  amplify_api: '<1.0.0'
```
3. Run `flutter pub get`

**Expected behavior**
Dependencies should resolve

**Platform**
Amplify Flutter current supports iOS and Android.  This issue is reproducible in (check all that apply):
[x] Android
[x] iOS

**Additional context**
Testing with Flutter stable 2.0

**Comment by haverchuck** [comment]
@fotiDim It appears that we are compatible with `image 2.x`, and are working on resolving this collision for `3.x`.

**Comment by fjnoyp** [comment]
Hi @fotiDim thanks for posting this issue. What haverchuck says is correct, downgrading to `image 2.x` will fix your issue for now.  

In order for us to resolve our collision with `3.x` it looks like we'll need to update our packages to be built with the latest version of flutter `2.*`.

**In more detail:** 
Currently you're getting an issue because our packages rely on `uuid 2.*`.  Unfortunately `uuid 3.*` and `image 3.x` rely on `crypto 3.x` which is not compatible with `flutter   Upgrading to  will not fix your issue because it relies on `cryto 3.x` which is not compatible with `flutter 1.*` which is what our Flutter plugins use.  

**Resolution:**
We are actively investigating upgrading the flutter version our Flutter plugins use to `flutter 2.*` and understand its importance as more people upgrade to this latest version of Flutter.  I can provide more details to you as that investigation continues.

**Comment by fjnoyp** [comment]
Hi @fotiDim!  We've found another way to fix your issue by using dependency_override. 

If you want to keep all of your existing dependencies as is, you can add the following to your pubspec.yaml: 

dependency_overrides:
  uuid: '3.0.1'

After making that change running `flutter pub get` should no longer result in the error that you've been seeing before.

**Comment by fotiDim** [comment]
Hi @fjnoyp. Thanks for the tip! Actually we had tried this already. Our variation was like this:
```
dependency_overrides:
  crypto: '2.1.5'
```
It was enough to resolve the dependency version conflicts. However right after we ran into compilation issues with `amplify_api` and `Flutter 2`. We were not sure if this was because of the overridden dependency or something else.

Eventually I managed to resolve the compilation issues using the workaround mentioned [here] (https://github.com/aws-amplify/amplify-flutter/issues/359#issuecomment-790052655). Maybe this should be mentioned in the documentation.

**Comment by haverchuck** [comment]
@fotiDim There is a fix for this pending.

**Comment by haverchuck** [comment]
@fotiDim A fix for this should be included in the 0.1.1 release (March 29).  Are you continuing to experience issues with this patch?

**Comment by fotiDim** [comment]
@haverchuck yes it is indeed resolved. Thank you!
